reana_workflow_controller/tasks.py

```diff
@@ -99,3 +105,24 @@ def _update_job_progress(workflow_uuid, msg):
                     run_job.run_id = current_run.id_
                     run_job.job_id = job_id
                     Session.add(run_job)
+
+
+def _update_job_cache(msg):
+    """Update caching information for finished job."""
+    cmd = msg['caching_info']['job_spec']['cmd']
+    # removes cd to workspace, to be refactored
+    clean_cmd = cmd.split(';')[1]
+    msg['caching_info']['job_spec']['cmd'] = clean_cmd
+    input_hash = calculate_job_input_hash(msg['caching_info']['job_spec'],
+                                          msg['caching_info']['workflow_json'])
+    directory_hash = calculate_hash_of_dir(
+        msg['caching_info'].get('workflow_workspace').
+        replace('data', 'reana/default'))
```

Comment:
This is going to invalidate the cache for all non `default` organization jobs. Something like https://github.com/reanahub/reana-job-controller/pull/74/commits/a2a73e6bf01f708b3b6d27fc387df569c09d8a7d needs to be done. You can get the experiment/organization name from the `job_spec`, we have to be careful though since soon we will be changing `experiment` variable name to `organization`.